.github/workflows/testAndPublish.yml

```
@@ -0,0 +1,248 @@
+      uses: actions/cache@v4
+      with:
+        path: ${{ github.workspace }}
+        key: ${{ github.ref }}-${{ github.run_id }}
+        fail-on-cache-miss: true
+    - name: Run unit tests
+      run: ci/scripts/tests/unitTests.ps1
+    - name: Upload artifact
+      id: uploadArtifact
+      if: ${{ always() }}
+      uses: actions/upload-artifact@v4
+      with:
+        name: Unit tests results
+        path: testOutput/unit/unitTests.xml
+        overwrite: true
+
+  createLauncher:
+    name: Create launcher
+    runs-on: windows-latest
+    needs: buildNVDA
+    steps:
+    - name: Checkout cached build
+      uses: actions/cache@v4
+      with:
+        path: ${{ github.workspace }}
+        key: ${{ github.ref }}-${{ github.run_id }}
+        fail-on-cache-miss: true
+    - name: Set version variables
+      run: ci/scripts/setBuildVersionVars.ps1
+    - name: Set scons args
+      run: ci/scripts/setSconsArgs.ps1
+    - name: Build launcher
+      run: scons launcher %sconsArgs% ${{ !runner.debug && '--all-cores' || '-j0'}}
+      shell: cmd
+    - name: Upload launcher
+      id: uploadLauncher
+      uses: actions/upload-artifact@v4
+      with:
+        name: NVDA launcher
+        path: output/nvda*.exe
+        overwrite: true
+
+  systemTests:
+    name: Run system tests
+    runs-on: windows-latest
+    needs: createLauncher
+    steps:
+    - name: Checkout cached build
+      uses: actions/cache@v4
+      with:
+        path: ${{ github.workspace }}
+        key: ${{ github.ref }}-${{ github.run_id }}
+        fail-on-cache-miss: true
+    - name: Get NVDA launcher
+      id: getLauncher
+      uses: actions/download-artifact@v4
+      with:
+        name: NVDA launcher
+        path: output
+    - name: Install NVDA
+      run: ci/scripts/installNVDA.ps1
+      env:
+        nvdaLauncherDir: ${{ steps.getLauncher.outputs.download-path }}
+    - name: Run system tests
+      run: ci/scripts/tests/systemTests.ps1
+      env:
+        nvdaLauncherDir: ${{ steps.getLauncher.outputs.download-path }}
+    - name: Upload system tests results
+      id: uploadSystemTestsResults
+      if: ${{ always() }}
+      uses: actions/upload-artifact@v4
+      with:
+        name: System tests results
+        path: testOutput/system
+
+  cleanupCache:
+    # Cache deletion on pull_request events cannot occur on PRs from forks
+    # as PRs from forks do not get write permissions from pull_request events.
+    # Alternatively, we can change the event trigger to pull_request_target,
+    # however that might introduce security risks as it grants fork PRs greater permissions.
+    # In these cases we can always let GitHub handle cache deletion:
+    # auto deletion after 7 days or limits are hit
+    name: Cleanup cache
+    permissions:
+      actions: write
+    needs: [checkPot, licenseCheck, unitTests, systemTests]
+    if: ${{ always() && (github.event.push || github.event.pull_request.head.repo.owner == 'nvaccess') }}
```

I don't think this workaround considers that strictly spoken, you are allowed to create a pr on your fork itself. Therefore this shouldn't be hardcoded to nvaccess, but rather to the owner of the current repository.
This ensure that for example, when I create a pr on my fork against my forked master, the cache will be deleted properly.
I think this is a legitimate way to test actions before a pull request is filed against the main repository, though workflow dispatch might be better in that case.
```suggestion
    if: ${{ always() && (github.event.push || github.event.pull_request.head.repo.owner == github.repository_owner) }}
```

---

yep this is fine - nice catch